plugins/module_utils/service_now.py

```diff
@@ -18,82 +20,469 @@
 except ImportError:
     PYSNOW_IMP_ERR = traceback.format_exc()
 
+# Pull in requests
+HAS_REQUESTS = False
+REQUESTS_IMP_ERR = None
+try:
+    import requests
+    HAS_REQUESTS = True
+except ImportError:
+    REQUESTS_IMP_ERR = traceback.format_exc()
+
 
-class ServiceNowClient(object):
-    def __init__(self, module):
-        """
-        Constructor
-        """
-        if not HAS_PYSNOW:
-            module.fail_json(msg=missing_required_lib('pysnow'), exception=PYSNOW_IMP_ERR)
-
-        self.module = module
-        self.params = module.params
-        self.client_id = self.params['client_id']
-        self.client_secret = self.params['client_secret']
-        self.username = self.params['username']
-        self.password = self.params['password']
-        self.instance = self.params['instance']
-        self.host = self.params['host']
-        self.session = {'token': None}
-        self.conn = None
-
-    def login(self):
-        result = dict(
-            changed=False
+class HTTPBearerAuth(requests.auth.AuthBase):
```

Comment:
`requests` needs to be conditional based on whether requests can be imported or not

Reply:
Once I conditionally defined HTTPBearerAuth, the remaining issues fell quickly. Thanks!